VolleyPaint/Assets/Scripts/Player/Movement/PlayerMovement.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.Netcode;
+
+public class PlayerMovement : NetworkBehaviour
+{
+	public float mouseSensitivityX = 1.0f;
+	public float mouseSensitivityY = 1.0f;
+
+	public float walkSpeed = 10.0f;
+	public bool canMove;
+
+	Vector3 moveAmount;
+	Vector3 smoothMoveVelocity;
+
+	Transform cameraT;
+	float verticalLookRotation;
+
+	Rigidbody rigidbodyR;
+
+	float jumpForce = 150.0f;
+	bool grounded;
+	public LayerMask groundedMask;
+	bool cursorVisible;
+
+	// Use this for initialization
+	void Start()
+	{
+		cameraT = Camera.main.transform;
+		rigidbodyR = GetComponent<Rigidbody>();
+		LockMouse();
+	}
+	public override void OnNetworkSpawn()
+	{
+		if (!IsOwner)
+		{
+			transform.Find("Main Camera").gameObject.GetComponent<Camera>().enabled = false;
+			transform.Find("Main Camera").gameObject.GetComponent<AudioListener>().enabled = false;
+			this.enabled = false;
+		}
+	}
+
+	// Update is called once per frame
+	void Update()
+	{
+		// rotation
+		transform.Rotate(Vector3.up * Input.GetAxis("Mouse X") * mouseSensitivityX);
+		verticalLookRotation += Input.GetAxis("Mouse Y") * mouseSensitivityY;
+		verticalLookRotation = Mathf.Clamp(verticalLookRotation, -60, 60);
+		cameraT.localEulerAngles = Vector3.left * verticalLookRotation;
+
+		// movement
+		if (canMove)
+        {
+			Vector3 moveDir = new Vector3(Input.GetAxisRaw("Horizontal"), 0, Input.GetAxisRaw("Vertical")).normalized;
+			Vector3 targetMoveAmount = moveDir * walkSpeed;
+			moveAmount = Vector3.SmoothDamp(moveAmount, targetMoveAmount, ref smoothMoveVelocity, .15f);
+
+        }
+
+		// jump
+		if (Input.GetButtonDown("Jump"))
+		{
+			if (grounded)
+			{
+				rigidbodyR.AddForce(transform.up * jumpForce);
+			}
+		}
+
+		Ray ray = new Ray(transform.position, -transform.up);
+		RaycastHit hit;
+
+		if (Physics.Raycast(ray, out hit, 1 + .1f, groundedMask))
+		{
+			grounded = true;
+		}
+		else
+		{
+			grounded = false;
+		}
+
+        /* Lock/unlock mouse on click */
+        if (Input.GetKeyDown(KeyCode.F1))
```

Comment:
Usually games use the escape key to enable your cursor while also bringing up a pause menu which we can talk about later about creating